docs/guides/developer/docs/installation/source-linux.md

```diff
@@ -43,45 +81,301 @@ install them.
+
+- ~/Projects/opencast 
+- /opt/activemq
+
+With that folder structure you could use following command:
+
+    $ cd && cd Projects && sudo cp opencast/docs/scripts/activemq/activemq.xml /opt/activemq/conf/activemq.xml
+
+If your folder structure is different from that example or you do decide to put it somewhere else, you should copy and replace the preconfigured XML from
+
+- /location/to/your/opencast/docs/scripts/activemq/activemq.xml
 
-4.Build and Start Opencast
+into
+
+- /location/to/your/activemq/conf/activemq.xml
+
+You can start your ActiveMQ instance with:
+
+    $ sudo ./location/to/your/activemq/bin/activemq start
+
+<br/>
+
+## Build and Start Opencast
 --------------------
 
-The -Pdev argument decreases the build time and skips the creation of multiple
-tarballs and turning on the developer tarball.
+You can build now opencast by changing your directory into your opencast location and by running:
+
+    $ mvn clean install
+
+After the successfully compilation you can start opencast with:
+
+    $ cd build/opencast-dist-develop-*/bin && ./start-opencast
+
+The -Pdev argument decreases the build time and skips the creation of multiple tarballs and turning on the developer tarball.
 
-    cd opencast-dir
-    mvn clean install -Pdev
-    cd build/opencast-*
-    ./bin/start-opencast
+    $ cd opencast && mvn clean install -Pdev
+    $ cd build/opencast-dist-develop-*/bin && ./start-opencast
 
 ### Useful Commands for Testing Purposes
 
 For a quick build, you can use the following command to skip Opencast's tests.
 
-    cd opencast-dir
-    mvn clean install -DskipTests=true
+    $ cd opencast
+    $ mvn clean install -Pdev -DskipTests=true
+
+To see the whole stacktrace of the installation you can use the following command to disable the trimming.
+
+    $ cd opencast
+    $ mvn clean install -DtrimStackTrace=false
+
+If you want to start opencast in debug mode, you could use the debug argument:
+
+    $ cd build/opencast-dist-develop-*/bin && ./start-opencast debug
+
+<br/>
+
+## Modify Code and Build Changes
+--------------------
+### After you modified your code you can go back to step "Build and Start Opencast" to rebuild Opencast.
+
+<br/>
 
-To see the whole stacktrace of the installation you can use the following command
-to disable the trimming.
+## Common Build Errors or Fixes
+--------------------
+
+### NPM Access Error
+
+If you get later on an npm access error, you can fix it by:
+
+    $ sudo chown -R $USER:$(id -gn $USER) ~/.config && sudo chown -R $USER:$(id -gn $USER) ~/.npm
+
+### JDK Version
+
+Make sure, that you use the 1.8.0 JDK, some IDE's could use a more up to date version, even if you set your system JDK to 1.8.0.
+
+### Waiting for ActiveMQ
+
+Make sure, that ActiveMQ runs without errors and with the right JAVA_HOME Variable (explained under section Ubuntu). Otherwise opencast will wait forever to connect.
```

Comment:
Opencast requires ActiveMQ to be both running and properly configured, otherwise it will wait forever to connect.  See here (link to anchor) for details on how to configure ActiveMQ.